### PR TITLE
Fix Typos and Styling for LimitRange Documentation

### PR DESCRIPTION
**Pull request by danielhelfand**
While reading through the documentation on Limit Range, I noticed several grammar and styling issues for the english version of the documentation. This pull request addresses many of these issues.

Of curiosity to me is if there are standards for how to refer to resources? For instance, pod and container are some times capitalized and other times not while namespace is always lowercase. Limit Range is also some times two words and some times one while also being capitalized and lowercased without consistency. Would be more than happy to address this if there are suggestions for this.

### CODE REVIEW

File: content/en/docs/concepts/policy/limit-range.md [hunk 1]
```diff
@@ -37,28 +37,28 @@ A limit range is enforced in a particular namespace when there is a
...
+- The `LimitRanger` admission controller enforces defaults and limits for all Pods and Container that do not set compute resource requirements and tracks usage to ensure it does not exceed resource minimum, maximum and ratio defined in any `LimitRange` present in the namespace.
+- If creating or updating a resource (Pod, Container, PersistentVolumeClaim) that violates a limit range  constraint, the request to the API server will fail with an HTTP status code `403 FORBIDDEN` and a message explaining the constraint that have been violated.
+- If a limit range is activated in a namespace for compute resources like `cpu` and `memory`, users must specify
+  requests or limits for those values. Otherwise, the system may reject pod creation.
+- Limit Range validations occurs only at Pod Admission stage, not on Running pods.
 
 
 Examples of policies that could be created using limit range are:
 
-- In a 2 node cluster with a capacity of 8 GiB RAM, and 16 cores, constrain Pods in a namespace to request 100m and not exceeds 500m for CPU , request 200Mi and not exceed 600Mi
```
Comment (danielhelfand) [inline content/en/docs/concepts/policy/limit-range.md]: The portion of the example here for `request 200Mi and not exceed 600Mi` did not make sense so I removed it. Is it referring to memory specifications for the limit range?

File: content/en/docs/concepts/policy/limit-range.md [hunk 2]
```diff
@@ -9,8 +9,8 @@ weight: 10
 {{% capture overview %}}
 
 By default, containers run with unbounded [compute resources](/docs/user-guide/compute-resources) on a Kubernetes cluster.
-With Resource quotas, cluster administrators can restrict the resource consumption and creation on a namespace basis.
-Within a namespace, a Pod or Container can consume as much CPU and memory as defined by the namespace's resource quota. There is a concern that one Pod or Container could monopolize all of the resources. Limit Range is a policy to constrain resource  by Pod or Container in a namespace.
+With resource quotas, cluster administrators can restrict resource consumption and creation on a namespace basis.
+Within a namespace, a Pod or Container can consume as much CPU and memory as defined by the namespace's resource quota. There is a concern that one pod or container could monopolize all available resources. A Limit Range is a policy to constrain resource consumption by Pod or Container in a namespace.
```
Comment (sftim) [inline content/en/docs/concepts/policy/limit-range.md]: Did you mean:
```suggestion
Within a namespace, a Pod or Container can consume as much CPU and memory as defined by the namespace's resource quota. There is a concern that one pod or container could monopolize all available resources. A LimitRange is a policy to constrain resource consumption by Pod or Container in a namespace.
```

or maybe:
```suggestion
Within a namespace, a Pod or Container can consume as much CPU and memory as defined by the namespace's resource quota. There is a concern that one pod or container could monopolize all available resources. A _limit range_ is a policy to constrain resource consumption by Pod or Container in a namespace.
```
?

File: content/en/docs/concepts/policy/limit-range.md [hunk 3]
```diff
@@ -37,28 +37,28 @@ A limit range is enforced in a particular namespace when there is a
...
 - Users create resources like Pods, Containers, and PersistentVolumeClaims in the namespace.
-- The `LimitRanger` admission controller enforces defaults  limits for all Pods and Container that do not set compute resource requirements and tracks usage to ensure it does not exceed resource minimum , maximum and ratio defined in any  `LimitRange` present in the namespace.
-- If creating or updating a resource (Pod, Container, PersistentVolumeClaim) violates a limit range  constraint, the request to the API server will fail with HTTP status code `403 FORBIDDEN` and a message explaining the constraint that would have been violated.
-- If limit range is activated in a namespace for compute resources like `cpu` and `memory`, users must specify
-  requests or limits for those values; otherwise, the system may reject pod creation.
-- LimitRange validations occurs only at Pod Admission stage, not on Running pods.
+- The `LimitRanger` admission controller enforces defaults and limits for all Pods and Container that do not set compute resource requirements and tracks usage to ensure it does not exceed resource minimum, maximum and ratio defined in any `LimitRange` present in the namespace.
```
Comment (sftim) [inline content/en/docs/concepts/policy/limit-range.md]: To follow the style guidelines, this could become:
```suggestion
- The `LimitRanger` admission controller enforces defaults and limits for all Pods and Containers that do not set compute resource requirements and tracks usage to ensure it does not exceed resource minimum, maximum and ratio defined in any LimitRange present in the namespace.
```

File: content/en/docs/concepts/policy/limit-range.md (hunk 1 above)
Comment (sftim) [reply content/en/docs/concepts/policy/limit-range.md]: > request 200Mi and not exceed 600Mi

I take that to mean that Pods will have a 200MiB memory request and to require that matched Pods have a memory limit at or below 600MiB.

File: content/en/docs/concepts/policy/limit-range.md [hunk 4]
```diff
@@ -37,28 +37,28 @@ A limit range is enforced in a particular namespace when there is a
...
 
-Neither contention nor changes to limitrange will affect already created resources.
+Neither contention nor changes to a limit range will affect already created resources.
 
 ## Limiting Container compute resources
 
-The following section discusses the creation of a LimitRange acting at Container Level.
-A Pod with 04 containers is first created; each container within the Pod has a specific `spec.resource` configuration
-each container within the pod is handled differently by the LimitRanger admission controller.
+The following section discusses the creation of a Limit Range acting at Container Level.
```
Comment (sftim) [inline content/en/docs/concepts/policy/limit-range.md]: The resource is called LimitRange; the original line is correct.

File: content/en/docs/concepts/policy/limit-range.md (hunk 2 above)
Comment (danielhelfand) [reply content/en/docs/concepts/policy/limit-range.md]: Yeah, I think that's the question I have is. It's referenced several ways in the doc. I think `LimitRange`, `Limit Range`, or `limit range` would be acceptable, but I'll review the style guide you have provided.

File: content/en/docs/concepts/policy/limit-range.md (hunk 1 above)
Comment (danielhelfand) [reply content/en/docs/concepts/policy/limit-range.md]: Thanks! Let me add it back/make it more clear.

File: content/en/docs/concepts/policy/limit-range.md [hunk 5]
```diff
@@ -1,64 +1,64 @@
...
+With resource quotas, cluster administrators can restrict resource consumption and creation on a namespace basis.
+Within a namespace, a Pod or Container can consume as much CPU and memory as defined by the namespace's resource quota. There is a concern that one Pod or Container could monopolize all available resources. A LimitRange is a policy to constrain resource consumption by Pod or Container in a namespace.
 
 {{% /capture %}}
 
 
 {{% capture body %}}
 
-A limit range, defined by a `LimitRange` object, provides constraints that can:
+A LimitRange, defined by a `LimitRange` object, provides constraints that can:
```
Comment (sftim) [inline content/en/docs/concepts/policy/limit-range.md]: This seems tautological. How about:
```suggestion
A _LimitRange_ provides constraints that can:
```

File: content/en/docs/concepts/policy/limit-range.md [hunk 6]
```diff
@@ -1,64 +1,64 @@
...
+## Enabling LimitRange
 
-Limit Range support is enabled by default for many Kubernetes distributions.  It is
+LimitRange support is enabled by default for many Kubernetes distributions. It is
 enabled when the apiserver `--enable-admission-plugins=` flag has `LimitRanger` admission controller as
 one of its arguments.
 
-A limit range is enforced in a particular namespace when there is a
+A LimitRange is enforced in a particular namespace when there is a
 `LimitRange` object in that namespace.
```
Comment (sftim) [inline content/en/docs/concepts/policy/limit-range.md]: Let's also fix this:
```suggestion
LimitRange object in that namespace.
```

File: content/en/docs/concepts/policy/limit-range.md [hunk 7]
```diff
@@ -1,64 +1,64 @@
...
 - Users create resources like Pods, Containers, and PersistentVolumeClaims in the namespace.
-- The `LimitRanger` admission controller enforces defaults  limits for all Pods and Container that do not set compute resource requirements and tracks usage to ensure it does not exceed resource minimum , maximum and ratio defined in any  `LimitRange` present in the namespace.
-- If creating or updating a resource (Pod, Container, PersistentVolumeClaim) violates a limit range  constraint, the request to the API server will fail with HTTP status code `403 FORBIDDEN` and a message explaining the constraint that would have been violated.
-- If limit range is activated in a namespace for compute resources like `cpu` and `memory`, users must specify
-  requests or limits for those values; otherwise, the system may reject pod creation.
-- LimitRange validations occurs only at Pod Admission stage, not on Running pods.
+- The `LimitRanger` admission controller enforces defaults and limits for all Pods and Containers that do not set compute resource requirements and tracks usage to ensure it does not exceed resource minimum, maximum and ratio defined in any `LimitRange` present in the namespace.
```
Comment (sftim) [inline content/en/docs/concepts/policy/limit-range.md]:
```suggestion
- The `LimitRanger` admission controller enforces defaults and limits for all Pods and Containers that do not set compute resource requirements and tracks usage to ensure it does not exceed resource minimum, maximum and ratio defined in any LimitRange present in the namespace.
```
LimitRange is a resource.

File: content/en/docs/concepts/policy/limit-range.md [hunk 8]
```diff
@@ -1,64 +1,64 @@
...
-Neither contention nor changes to limitrange will affect already created resources.
+Neither contention nor changes to a LimitRange will affect already created resources.
 
 ## Limiting Container compute resources
 
 The following section discusses the creation of a LimitRange acting at Container Level.
-A Pod with 04 containers is first created; each container within the Pod has a specific `spec.resource` configuration
-each container within the pod is handled differently by the LimitRanger admission controller.
+A Pod with 04 containers is first created. Each container within the Pod has a specific `spec.resource` configuration.
+Each container within the pod is handled differently by the LimitRanger admission controller.
```
Comment (sftim) [inline content/en/docs/concepts/policy/limit-range.md]:
```suggestion
Each container within the pod is handled differently by the `LimitRanger` admission controller.
```

File: content/en/docs/concepts/policy/limit-range.md [hunk 9]
```diff
@@ -125,9 +125,9 @@ kubectl get po/busybox1 -n limitrange-demo -o json | jq ".spec.containers[0].res
 ```
 
 - The `busybox-cnt01` Container inside `busybox` Pod defined `requests.cpu=100m` and `requests.memory=100Mi`.
--  `100m <= 500m <= 800m` , The container cpu limit (500m) falls inside the authorized CPU limit range.
--  `99Mi <= 200Mi <= 1Gi` , The container memory limit (200Mi) falls inside the authorized Memory limit range.
--  No request/limits ratio validation for  CPU/Memory , thus the container is valid and created.
```
Comment (sftim) [inline content/en/docs/concepts/policy/limit-range.md]: I'm happy with the existing “limit range” wording here.

File: content/en/docs/concepts/policy/limit-range.md [hunk 10]
```diff
@@ -151,14 +151,15 @@ kubectl get po/busybox1 -n limitrange-demo -o json | jq ".spec.containers[1].res
 }
 ```
 -  The `busybox-cnt02` Container inside `busybox1` Pod defined `requests.cpu=100m` and `requests.memory=100Mi` but not limits for cpu and memory.
--  The container do not have a limits section, the default limits defined in the limit-mem-cpu-per-container LimitRange object are injected to this container `limits.cpu=700mi` and `limits.memory=900Mi`.
--  `100m <= 700m <= 800m` , The container cpu limit (700m) falls inside the authorized CPU limit range.
--  `99Mi <= 900Mi <= 1Gi` , The container memory limit (900Mi) falls inside the authorized Memory limit range.
--  No request/limits ratio  set , thus the container is valid and created.
```
Comment (sftim) [inline content/en/docs/concepts/policy/limit-range.md]: I'm happy with the existing “limit range” wording here.

File: content/en/docs/concepts/policy/limit-range.md [hunk 11]
```diff
@@ -177,10 +178,10 @@ kubectl get po/busybox1 -n limitrange-demo -o json | jq ".spec.containers[2].res
 ```
 
 -  The `busybox-cnt03` Container inside `busybox1` Pod defined `limits.cpu=500m` and `limits.memory=200Mi` but no `requests` for cpu and memory.
--  The container do not define a request section, the defaultRequest defined in the limit-mem-cpu-per-container LimitRange is not used to fill its limits section but the limits defined by the container are set as requests `limits.cpu=500m` and `limits.memory=200Mi`.
--  `100m <= 500m <= 800m` , The container cpu limit (500m) falls inside the authorized CPU limit range.
--  `99Mi <= 200Mi <= 1Gi` , The container memory limit (200Mi) falls inside the authorized Memory limit range.
--  No request/limits ratio  set , thus the container is valid and created.
```
Comment (sftim) [inline content/en/docs/concepts/policy/limit-range.md]: I'm happy with the existing “limit range” wording here.

File: content/en/docs/concepts/policy/limit-range.md [hunk 12]
```diff
@@ -1,64 +1,64 @@
...
 weight: 10
 ---
 
 {{% capture overview %}}
 
 By default, containers run with unbounded [compute resources](/docs/user-guide/compute-resources) on a Kubernetes cluster.
-With Resource quotas, cluster administrators can restrict the resource consumption and creation on a namespace basis.
-Within a namespace, a Pod or Container can consume as much CPU and memory as defined by the namespace's resource quota. There is a concern that one Pod or Container could monopolize all of the resources. Limit Range is a policy to constrain resource  by Pod or Container in a namespace.
+With resource quotas, cluster administrators can restrict resource consumption and creation on a namespace basis.
+Within a namespace, a Pod or Container can consume as much CPU and memory as defined by the namespace's resource quota. There is a concern that one Pod or Container could monopolize all available resources. A LimitRange is a policy to constrain resource consumption by Pod or Container in a namespace.
```
Comment (sftim) [inline content/en/docs/concepts/policy/limit-range.md]: How about: “A LimitRange is a policy to constrain resource allocations (to Pods or Containers) in a namespace.”

One thing I want readers to learn is that LimitRanger can enforce _minimum_ request (and _minimum_ limit). So if I want to create a Pod that requests 5m CPU and 16KiB memory, someone who manages that namespace can force me to request more CPU and more memory. I'm thus restricted in my ability to mislead the scheduler.

File: content/en/docs/concepts/policy/limit-range.md (hunk 9 above)
Comment (danielhelfand) [reply content/en/docs/concepts/policy/limit-range.md]: So question on this: I don't see limit range as two separate words in other docs on the website. It always seems to be LimitRange. Why would this be an exception?

Comment (sftim) [reply content/en/docs/concepts/policy/limit-range.md]: Here, “limit range” is used to mean “a range of valid values for some subfield(s) of `requests` or `limits`” whereas elsewhere “LimitRange” means “the Kubernetes API object that represents a constraint on acceptable values for some subfield(s) of `requests` or `limits`” .

The name LimitRange is kind of unhelpful because it implies it's a constraint specifically on `limits`, rather than being applicable to both `requests` and `limits`.

Comment (danielhelfand) [reply content/en/docs/concepts/policy/limit-range.md]: Ah, tricky! I agree, and it makes sense. I'll change back this and the others you have highlighted. Thank you.

File: content/en/docs/concepts/policy/limit-range.md [hunk 13]
```diff
@@ -1,64 +1,64 @@
...
 content_template: templates/concept
 weight: 10
 ---
 
 {{% capture overview %}}
 
-By default, containers run with unbounded [compute resources](/docs/user-guide/compute-resources) on a Kubernetes cluster.
-With Resource quotas, cluster administrators can restrict the resource consumption and creation on a namespace basis.
-Within a namespace, a Pod or Container can consume as much CPU and memory as defined by the namespace's resource quota. There is a concern that one Pod or Container could monopolize all of the resources. Limit Range is a policy to constrain resource  by Pod or Container in a namespace.
+By default, Containers run with unbounded [compute resources](/docs/user-guide/compute-resources) on a Kubernetes cluster.
```
Comment (sftim) [inline content/en/docs/concepts/policy/limit-range.md]: I prefer the original line. The term “container” here is about the general IT concept rather than an element of Kubernetes' API.

File: content/en/docs/concepts/policy/limit-range.md [hunk 14]
```diff
@@ -1,64 +1,64 @@
 ---
 reviewers:
 - nelvadas
-title: Limit Ranges
```
Comment (sftim) [inline content/en/docs/concepts/policy/limit-range.md]: The original title, “Limit Ranges”, should localize into target languages. I like that it would.
How about changing the title to “Resource Limit Ranges”?

File: content/en/docs/concepts/policy/limit-range.md [hunk 15]
```diff
@@ -125,9 +125,9 @@ kubectl get po/busybox1 -n limitrange-demo -o json | jq ".spec.containers[0].res
 ```
 
 - The `busybox-cnt01` Container inside `busybox` Pod defined `requests.cpu=100m` and `requests.memory=100Mi`.
--  `100m <= 500m <= 800m` , The container cpu limit (500m) falls inside the authorized CPU limit range.
--  `99Mi <= 200Mi <= 1Gi` , The container memory limit (200Mi) falls inside the authorized Memory limit range.
--  No request/limits ratio validation for  CPU/Memory , thus the container is valid and created.
+-  `100m <= 500m <= 800m` , The Container cpu limit (500m) falls inside the authorized CPU LimitRange.
+-  `99Mi <= 200Mi <= 1Gi` , The Container memory limit (200Mi) falls inside the authorized Memory LimitRange.
+-  No request/limits ratio validation for CPU/Memory, so the Container is valid and will be created.
```
Comment (sftim) [inline content/en/docs/concepts/policy/limit-range.md]: Can you rewrite this without “will be”? Usually that's possible, and more in line with the style guide.

(You might have to change more of the text to achieve that).

File: content/en/docs/concepts/policy/limit-range.md [hunk 16]
```diff
@@ -328,9 +329,9 @@ Error from server (Forbidden): error when creating "pvc-limit-greater.yaml": per
 
 ## Limits/Requests Ratio
 
-If `LimitRangeItem.maxLimitRequestRatio` is specified in the `LimitRangeSpec`, the named resource must have a request and limit that are both non-zero where limit divided by request is less than or equal to the enumerated value
+If `LimitRangeItem.maxLimitRequestRatio` is specified in the `LimitRangeSpec`, the named resource must have a request and limit that are both non-zero where limit divided by request is less than or equal to the enumerated value.
 
-The following `LimitRange` enforces memory limit to be at most twice the amount of the memory request for any pod in the namespace.
+The following LimitRange enforces memory limit to be at most twice the amount of the memory request for any pod in the namespace:
```
Comment (sftim) [inline content/en/docs/concepts/policy/limit-range.md]: nit, how about:
```suggestion
The following LimitRange enforces memory limit to be at most twice the amount of the memory request for any Pod in the namespace:
```
?

File: content/en/docs/concepts/policy/limit-range.md [hunk 17]
```diff
@@ -352,15 +353,15 @@ Type        Resource  Min  Max  Default Request  Default Limit  Max Limit/Reques
...
 {{< codenew file="admin/resource/limit-range-pod-3.yaml" >}}
 
 ```shell
 kubectl apply -f https://k8s.io/examples/admin/resource/limit-range-pod-3.yaml
 ```
 
-The pod creation failed as the ratio here (`3`) is greater than the enforced limit (`2`) in `limit-memory-ratio-pod` LimitRange
+The pod creation failed as the ratio here (`3`) is greater than the enforced limit (`2`) in `limit-memory-ratio-pod` LimitRange:
 
 ```shell
```
Comment (sftim) [inline content/en/docs/concepts/policy/limit-range.md]: Aside: this is output, it shouldn't be marked as shell.

@danielhelfand it's up to you if you want to fix that or leave it to a different PR.

File: content/en/docs/concepts/policy/limit-range.md (hunk 13 above)
Comment (danielhelfand) [reply content/en/docs/concepts/policy/limit-range.md]: Sure, I will return to lowercase. My fault. I didn't realize capitals were reserved for API concepts.

File: content/en/docs/concepts/policy/limit-range.md (hunk 14 above)
Comment (danielhelfand) [reply content/en/docs/concepts/policy/limit-range.md]: I guess I would prefer to just keep `Limit Ranges` in this case, but I agree on the rationale.

File: content/en/docs/concepts/policy/limit-range.md (hunk 15 above)
Comment (danielhelfand) [reply content/en/docs/concepts/policy/limit-range.md]: Removed it.

File: content/en/docs/concepts/policy/limit-range.md (hunk 16 above)
Comment (danielhelfand) [reply content/en/docs/concepts/policy/limit-range.md]: Sorry, missed this one.

File: content/en/docs/concepts/policy/limit-range.md (hunk 17 above)
Comment (danielhelfand) [reply content/en/docs/concepts/policy/limit-range.md]: I have updated it.

**Comment by sftim** [comment]
@danielhelfand would you be willing to rebase this against _master_? There may be some conflicts.

**Comment by danielhelfand** [comment]
> @danielhelfand would you be willing to rebase this against _master_? There may be some conflicts.

Rebased

**Comment by sftim** [comment]
[Preview page](https://deploy-preview-18961--kubernetes-io-master-staging.netlify.com/docs/concepts/policy/limit-range/)

I glanced at that and it looks OK. Right now I don't have time to review formally I'm afraid.
